dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/ClassLoadingServiceImpl.java

```diff
@@ -607,8 +606,12 @@ private void listenForLibraryChanges(String libid, AppClassLoader acl) {
             @Override
             protected void update() {
                 AppClassLoader cl = get();
-                if (cl != null && aclStore != null)
-                    aclStore.remove(cl);
+                if (cl != null) {
+                    if (aclStore != null) {
```

Comment:
`aclStore` looks like it should be `final`.  There is no point in creating it in `activate` and `null` on `deactivate`.  On `deactivate` the object is thrown away for GC anyway.  Nulling stuff out in `deactivate` is only useful if it is to represent some state of the deactivated object.  I don't think that is the case here.  It would be nice not to have to worry about this being `null`.  Seems `tcclStore` should also be final.

Reply:
I updated to make the fields final as suggested.